### DISCUSSION TITLE
Drop `ShieldCancellation`.

### DISCUSSION THREAD

**Discussion by T-256** [encode/httpcore#806] [General]
I didn't see any discussion about this, I think that's proper place to handle it in `unasync.py`.

**Comment by lovelydinosaur** [comment]
As with #805, can you describe and demonstrate a behaviour that you want to see changed?

Driving changes in the project should be through describing changes in terms of behaviour & motivation.

**Reply by T-256** [reply]
> Driving changes in the project should be through describing changes in terms of behaviour & motivation.

Sorry, `ShieldCancellation` is not member of public API, I thought I shouldn't consider it as _behavior change_.

The main motivation is when I was reading few files in `httpcore/_sync/`, I confused on the error handlings (task cancellation shield on non-async codes!?). Then by going to `httpcore/_synchronization.py`, I noticed that is only for prevent `unasync.py` errors.

**Reply by lovelydinosaur** [reply]
`ShieldCancellation` is a nop in the sync codebase.

There isn't any obvious gracefully way to approach "shielded cancellation blocks in our async code" that doesn't involve also mirroring that into the sync part of the codebase, so we make do with having a no-op context block there. If you have a look through the history you'd probably be able to find some discussion about that.
